src/util/StringUtils.h

```diff
@@ -536,53 +540,86 @@ inline string lstrip(const string& text, const char* s) {
+inline string rstrip(const string& text, const char c) {
+  auto it = text.end();
+  for (; it > text.begin(); it--) {
+    if (*(it - 1) != c) {
+      break;
+    }
   }
-  return text.substr(0, i);
+  return string(text.begin(), it);
 }
 
 // _____________________________________________________________________________
-inline string rstrip(const string& text, string s) {
+inline string rstrip(const string& text, const string& s) {
   // 256 bytes should fit on almost any stack
   // note the {}, it initializes with 0 = false
   array<bool, std::numeric_limits<unsigned char>::max() + 1> chars{};
-  for (size_t i = 0; i < s.size(); ++i) {
-    chars[s[i]] = true;
+  for (unsigned char c : s) {
+    chars[c] = true;
   }
-  size_t i = text.size();
-  while (i > 0 && chars[static_cast<unsigned char>(text[i - 1])]) {
-    --i;
+  auto it = text.end();
+  for (; it > text.begin(); it--) {
+    if (!chars[static_cast<unsigned char>(*(it - 1))]) {
+      break;
+    }
   }
-  return text.substr(0, i);
+  return string(text.begin(), it);
 }
 
 // _____________________________________________________________________________
 inline string rstrip(const string& text, const char* s) {
+  // We actually only implement the std::string variant. Over this variant it
+  // has the advantage to also work with embedded 0 bytes
   return rstrip(text, string(s));
 }
 
 // _____________________________________________________________________________
-inline std::string strip(const std::string& text, char c) {
+inline string strip(const std::string& text, const char c) {
   auto left = text.begin();
-  auto right = text.end();  // right is of course exclusive
-  for (; left != text.end() && *left == c; left++)
-    ;
-  for (; right != text.begin() && *(right - 1) == c; right--)
-    ;
-  return std::string(left, right);
+  auto right = text.end();
+  for (; left < text.end(); left++) {
+    if (*left != c) {
+      break;
+    }
+  }
+
+  for (; right > left; right--) {
+    if (*(right - 1) != c) {
+      break;
+    }
+  }
+  return string(left, right);
 }
 
 // _____________________________________________________________________________
-string strip(const string& text, const char* s) {
+inline string strip(const string& text, const char* s) {
+  // We actually only implement the std::string variant. Over this variant it
+  // has the advantage to also work with embedded 0 bytes
   return strip(text, string(s));
 }
 
 // _____________________________________________________________________________
-string strip(const string& text, const string& s) {
-  return rstrip(lstrip(text, s), s);
+inline string strip(const string& text, const string& s) {
+  // 256 bytes should fit on almost any stack
+  // note the {}, it initializes with 0 = false
+  array<bool, std::numeric_limits<unsigned char>::max() + 1> chars{};
+  for (unsigned char c : s) {
+    chars[c] = true;
+  }
+  auto left = text.begin();
+  auto right = text.end();
+  for (; left < text.end(); left++) {
```

Comment:
Here one could also use the `right` instead of `text.end()`.

Reply:
If the strip functions are used in a performance critical area, we could also try returning string_views, but I am not sure whether this causes too much trouble when explicitly having to cast them to strings.

Reply:
I think the main problem with the `string_view` approach is that we must be sure that none of the strings that are viewed goes out of scope/is deleted while we keep a view.